integration-tests/utils/integration-utils.js

```diff
@@ -211,9 +212,45 @@ const dismissNPSSurvey = async page => {
   }
 }
 
+// Test workaround: Retry loading of Terra UI if fails first time. This issue often happens after new deploy to Staging/Alpha.
 const signIntoTerra = async (page, { token, testUrl }) => {
-  !!testUrl && await page.goto(testUrl, navOptionNetworkIdle())
-  await page.waitForXPath('//*[contains(normalize-space(.),"Loading Terra")]', { hidden: true, timeout: 60 * 1000 })
+  console.log('signIntoTerra ...')
+
+  const retryOptions = {
+    factor: 1,
+    minTimeout: 1000, // This is min wait time between retries
+    onFailedAttempt: error => {
+      console.error(
+        `Sign in to Terra attempt ${error.attemptNumber} failed. There are ${error.retriesLeft} retries left.`
+      )
+    },
+    retries: 2
+  }
+
+  const timeout = 60 * 1000
+  const run = async url => {
+    try {
+      const httpResponse = await page.goto(url, { waitUntil: ['networkidle0', 'domcontentloaded'], timeout })
+      if (httpResponse.status() >= 400) {
+        throw new Error(`Error loading URL: ${url}. Http response status: ${httpResponse.statusText()}`)
+      }
+      await page.waitForFunction(url => {
+        return window.location.href.includes(url)
+      }, {}, url)
+      await page.waitForXPath('//*[contains(normalize-space(.),"Loading Terra")]', { hidden: true, timeout })
+    } catch (e) {
+      console.error(e)
```

Comment:
Will this error be logged twice? Once here and once in the `onFailedAttempt` function in `retryOptions`?

Reply:
I'm not sure because Sign In does not fail when I run tests locally and in CircleCI without deploy first. I removed `console.error(e)` for now.